src/main/java/core/basesyntax/serviceimpl/DataConverterImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverterService;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverterService {
+    private static final String SEPARATOR = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> convert(List<String> dataFromFile) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : dataFromFile) {
+            String[] data = line.split(SEPARATOR);
+            FruitTransaction.Operation operation = FruitTransaction
+                    .Operation.getOperationByCode(data[OPERATION_INDEX]);
+            String fruit = data[FRUIT_INDEX];
+            int quantity = Integer.parseInt(data[QUANTITY_INDEX]);
+            transactions.add(new FruitTransaction(operation, fruit, quantity));
+        }
+        return transactions;
+    }
```

Comment:
Please ensure the data conversion handles potential exceptions that might occur due to malformed input data. For instance, parsing an integer could throw a `NumberFormatException` if the input is not a valid integer. It is recommended to catch these exceptions and rethrow them with an informative message.